### ISSUE TITLE
ECNAnalysis never seems to get any results

### ISSUE THREAD

**Issue by britram** [mami-project/pathspider#12]
Running standalone with webresolver with about 100 targets, in the current testing branch (`new-mplane-final-demo`), running against the mPlane SDK in the `master` branch `mami-project/mplane-sdk`, ecnspider runs, but the result_sink method in EcnAnalysis never gets called.

Why? I'm completely lost in tracking this one down. 

Will need to fall back to the old demo as a plan next Monday at the latest (and I'm out of cycles on this 'til then).

**Comment by britram** [comment]
hm, this might have something to do with it:

```
Traceback (most recent call last):
  File "/usr/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/home/briant/src/pathspider/pathspider/client/ecnclient.py", line 420, in analyzer_func
    self.result_sink(analysis, chunk_id)
  File "/home/briant/src/pathspider/pathspider/__init__.py", line 615, in ecn_result_sink
    self.subjects_map[str(ip)]['ecn'] = status
KeyError: '5.9.142.248'
```

**Comment by britram** [comment]
debeb587be525e2b1ecb95bc49d547068ca4a936 and 938dc8f92e4bd2b8edae52de174428c192dca1fc fixed the previous crashes, but I still get null results everywhere.

**Comment by britram** [comment]
Synchronization issues fixed, but there's still a serialization issue:

```
Traceback (most recent call last):
  File "/home/briant/src/pathspider/pathspider/__init__.py", line 575, in perform
    json.dump({'subjects': self.subjects}, self.report_file, cls=IPAddressEncoder)
  File "/usr/lib/python3.4/json/__init__.py", line 178, in dump
    for chunk in iterable:
  File "/usr/lib/python3.4/json/encoder.py", line 422, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.4/json/encoder.py", line 396, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.4/json/encoder.py", line 317, in _iterencode_list
    yield from chunks
  File "/usr/lib/python3.4/json/encoder.py", line 396, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.4/json/encoder.py", line 429, in _iterencode
    o = _default(o)
  File "/home/briant/src/pathspider/pathspider/__init__.py", line 35, in default
    return json.JSONEncoder.default(self, obj)
  File "/usr/lib/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: Local:conn                       3
Local:nego                   False
destination.ip4     192.251.189.44
destination.port                80
Name: 192.251.189.44, dtype: object is not JSON serializable
```

**Comment by gubser** [comment]
fixed in aa41535817d4c9778e88c93b7530c16ad3943d70
